### ISSUE TITLE
Ncurses fails to install?

### ISSUE THREAD

**Issue by AwlsomeAlex** [ivandavidov/minimal#27]
Hello, I have been working recently on a separate branch of minimal to add support of Nano, then to push it into the master branch for everyone to use, but I'm having trouble installing Ncurses and putting the libraries in the /lib folder required by Nano. Like it will fail at a mouse problem and won't generate the libraries, with the worst part being the file that's broken builds the libraries. Think you could take a look at it and see if I might've typed something wrong, because I checked the Internet and used Linux From Scratch as a source and nothing seemed to be wrong. 
My Fork: [https://github.com/AwlsomeAlex/minimal](url)
(I know this has nothing to do with yours, but I wish to contribute to yours with Nano and eventually dialog being something I would like to implement.)

**Comment by ivandavidov** [comment]
Take a look at my code...

https://github.com/AwlsomeAlex/minimal/blob/master/src/overlay_02_build_links.sh

...and now take a look at your code...

https://github.com/AwlsomeAlex/minimal/blob/master/src/overlay_06_build_ncurses.sh

There is a difference - the right location to copy the libraries would be this one:

`$SRC_DIR/work/src/minimal_rootfs/lib`

That's because the folder `$SRC_DIR/minimal_rootfs` is used as a **static template** but the actual build-time artifacts have to go to `$SRC_DIR/work/src/minimal_rootfs/lib`. After that the ISO image generation process gets all files from `$SRC_DIR/work/src/minimal_rootfs` and puts them on the ISO image.

So, I suggest you go through your modifications and revise the directories that you use.

**Comment by AwlsomeAlex** [comment]
Okay, but also Ncurses will fail to build, without creating the libraries. Am I using the right ./configure options?

**Comment by ivandavidov** [comment]
The only way to answer that would be to build nano + ncurses myself and tell you how I built them. :)

The usual pattern for building additional software is:

1) Get the source bundle and extract it.
2) Find out how to do the initial configuration. Usually it's done by a `./configure` script.
3) Try to create a default configuration.
4) Try to make the software and then install it in separate folder.
5) Analyze the content of the installation folder and place all relevant files in the `minimal_overlay` folder. Side note - in my previous post I incorrectly told you that you need to put your stuff in `minimal_rootfs`. That's partially correct if you want your software to be visible in the initramfs space **before** the overlay process is triggered, while all files in `minimal_overlay` will be visible only **after** the overlay process is triggered.
6) Play with the configuration options and figure out what you need and what you don't need. Be advised that the overlay software dependencies usually are OS specific and it's up to you to resolve all such dependencies for your host machine.

Long story short - I can take a look at nano/ncurses when I have some free time but I don't promise anything. I'll leave the issue ticket open for now. Please update it if/when you have progress.

**Comment by AwlsomeAlex** [comment]
Okay. i'll experiment with the Ncurses config files, and email the developers the problem, as soon as I find a fix, i'll apply it and send a pull request, maybe then it will be useful for future programs. :)

**Comment by ivandavidov** [comment]
FYI - the commits from today (23-May) add all GNU C libraries as additional overlay bundle. This should solve half of the problem - if ncurses require glibc (glibc also requires glibc sometimes, e.g. NSS libraries), then these libraries are now present in the overlay structure.

**Comment by ivandavidov** [comment]
@AwlsomeAlex - is there any progress on this issue?

**Comment by AwlsomeAlex** [comment]
Yeah, I'm waiting for the next release with full Glibc Libraries, once it's released I will start the same process which I used to try building Ncurses and its Libraries, and several other methods. With the addition of Ncurses, you'll be able to expand the project with additions of Dialog/Text-Based Interfaces such as Nano and WPA_Supplicants/Wifi-Tools, so I'll try to push on development once a stable is released. @ivandavidov

**Comment by ivandavidov** [comment]
I'm closing this issue since it is no relevant for the current development cycle.
